.NET/Microsoft.Recognizers.Text.NumberWithUnit/Extractors/BaseMergedExtractor.cs

```diff
@@ -0,0 +1,174 @@
+                    continue;
+                }
+
+                // Separated by separators
+                var match = config.CompoundUnitRegex.Match(middleStr);
+                if (match.Success && match.Index == 0 && match.Length == middleStr.Length)
+                {
+                    groups[idx + 1] = groups[idx];
+                }
+                else
+                {
+                    groups[idx + 1] = groups[idx] + 1;
+                }
+            }
+
+            for (var idx = 0; idx < ers.Count; idx++)
+            {
+                if (idx == 0 || groups[idx] != groups[idx - 1])
+                {
+                    var tmpExtractResult = ers[idx];
+                    tmpExtractResult.Data = new List<ExtractResult>
+                    {
+                        new ExtractResult
+                        {
+                            Data = ers[idx].Data,
+                            Length = ers[idx].Length,
+                            Start = ers[idx].Start,
+                            Text = ers[idx].Text,
+                            Type = ers[idx].Type
+                        }
+                    };
+                    result.Add(tmpExtractResult);
+                }
+
+                // Reduce extract results in same group
+                if (idx + 1 < ers.Count && groups[idx + 1] == groups[idx])
+                {
+                    var group = groups[idx];
+
+                    var periodBegin = result[group].Start ?? 0;
+                    var periodEnd = (ers[idx + 1].Start ?? 0) + (ers[idx + 1].Length ?? 0);
+
+                    result[group].Length = periodEnd - periodBegin;
+                    result[group].Text = source.Substring(periodBegin, periodEnd - periodBegin);
+                    result[group].Type = Constants.SYS_UNIT_COMPOUND;
+                    (result[group].Data as List<ExtractResult>)?.Add(ers[idx + 1]);
+                }
+            }
+
+            for (var idx = 0; idx < result.Count; idx++)
+            {
+                var innerData = result[idx].Data as List<ExtractResult>;
+                if (innerData?.Count == 1)
+                {
+                    result[idx] = innerData[0];
+                }
+            }
+
+            return result;
+        }
+
+        private void MergePureNumber(string source, List<ExtractResult> ers)
+        {
+            var numErs = config.UnitNumExtractor.Extract(source);
+            var unitNumbers = new List<ExtractResult>();
+            for (int i = 0, j = 0; i < numErs.Count; i++)
+            {
+                bool hasBehindExtraction = false;
+                while (j < ers.Count && ers[j].Start + ers[j].Length < numErs[i].Start)
+                {
+                    hasBehindExtraction = true;
+                    j++;
+                }
+
+                if (!hasBehindExtraction) continue;
+
+                var middleBegin = ers[j - 1].Start + ers[j - 1].Length ?? 0;
```

Comment:
be careful of using ??, because of the priority of that. For that case, it means (ers[j - 1].Start + ers[j - 1].Length) ?? 0, is that right?

Reply:
Yeah, I think the `??` has a lower priority than `+`, so I simply write this.